pushnotifications/src/main/java/com/pusher/pushnotifications/reporting/FCMMessageReceiver.kt

```diff
@@ -25,8 +23,13 @@ class FCMMessageReceiver : WakefulBroadcastReceiver() {
     }
 
     intent?.getStringExtra("pusher")?.let { pusherDataJson ->
-      try {
-        val pusherData = gson.fromJson(pusherDataJson, PusherMetadata::class.java)
+        val pusherData = try {
+          gson.fromJson(pusherDataJson, PusherMetadata::class.java)
+      } catch (_: JsonSyntaxException) {
+          // TODO: Add client-side reporting
```

Comment:
Is this a WIP change in this branch, or does the TODO refer to some tech-debt that is being knowingly introduced now as a workaround? If it's the latter, then can you please track this by opening an issue discussing how this might be resolved later? If it's the former, then it should be added before the PR is merged.

Reply:
This was already present. It looks like a new change because of the reduced scope of this try-catch. This was here before: https://github.com/pusher/push-notifications-android/pull/120/files/478d00107e17e304754a24c8970fbd585c6eeede#diff-d9fc7bcb4783f1533ad8f0a4d1f76e6df88a0c689d7f95304689c583a25b7cd7L66